Feature/instruments

jakeschurch opened the pull request:
Add base Instrument types.

Added:

- [x] Instrument Trait
- [x] Order Type
- [x] Quote Type
- [x] Holding Type

Commit:
Add Currency traits

Commit:
Repackage repo & add Order struct semantics

Commit:
Introduce benchmarks

Commit:
feat(instruments): Add Currency type

- Add benchmarks
- Add test suite

GitCop commented:
There were the following issues with your Pull Request

 * Commit: 2d13e4f24a6c40bb7a6fd3556328bfb49c9cd218
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 62e36b325475fad2abb75c171d3c0e0d29db1dda
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 3b13d06bb95cf7a71ea1fbbae23411f665a94268
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/jakeschurch/trade_rs

---
This message was auto-generated by https://gitcop.com

jakeschurch committed:
Implement Holding functionality

- add Holding type
- add tests for mod.rs
- update dev dependencies
- other minor changes

Signed-off-by: Jake Schurch <jakeschurch@gmail.com>

jakeschurch commented:
NOTE: appveyor script not setup.
